### PR TITLE
Makes Redis session use a different Cache instance than app('cache')

### LINKED ISSUE #9157: Redis session changes cache connection
Like described in #7896, if both session and cache are configured to use redis, but different redis dbs or different redis instances, all `Cache::` operations go througth the connection used by `Session::`. This makes Cache configuration meaningless. 

If you think this is a problem and should be fixed, I will send a pull-request.

### PR DESCRIPTION
**Pull request by ifduyue**
So that Redis session doesn't bother with the connection hold by app('cache').

This PR fixes #9157

### CODE REVIEW

**Comment by ifduyue** [comment]
I have create a [repo](https://github.com/ifduyue/laravel-5-redis-session-demo) to reproduce the problem that this PR tries to fix.

php artisan serve and then curl http://localhost:8000/, and you'll see cache connection parameters are the same with session connection parameters.

File: src/Illuminate/Session/SessionManager.php [hunk 1]
```diff
@@ -125,7 +126,9 @@ protected function createWincacheDriver()
      */
     protected function createRedisDriver()
     {
-        $handler = $this->createCacheHandler('redis');
+        $minutes = $this->app['config']['session.lifetime'];
+        $cache = new CacheManager($this->app);
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/SessionManager.php]: This is wrong. You MUST resolve the cache manager from the ioc container.

Comment (ifduyue) [inline src/Illuminate/Session/SessionManager.php]: I pushed a new commit. 
`$cache = clone $this->app['cache'];` before `setConnection`, is this acceptable?

$this->app['cache'] is registered as a singleton, I can't think a better way than clone it.

**Comment by GrahamCampbell** [comment]
We should be using the same cache manager instance, and not cloning it. Looks like this "fix" does nothing.

**Comment by GrahamCampbell** [comment]
Specifying the driver to use, DOES give us the correct instance.

**Comment by franzliedke** [comment]
That's not what #9157 says, though...

**Comment by ifduyue** [comment]
@GrahamCampbell Specifying the driver gives us a correct instance, but it's configured according to the cache connection, not the session connection. After setConnection the problem mentioned above occurs.

**Comment by ifduyue** [comment]
> We should be using the same cache manager instance, and not cloning it. Looks like this "fix" does nothing.

It does something, since it's a different object, calling setConnection on it will not change the cache singleton's connection.

**Comment by ifduyue** [comment]
Here is a demo for this fix.

```php
Route::get('/', function () {
    $reader = function & ($object, $property) {
        $value = & Closure::bind(function & () use ($property) {
            return $this->$property;
        }, $object, $object)->__invoke();

        return $value;
    };

    return [
        'config' => config('database.redis'),
        'cache\'s connection parameters' => $reader($reader($reader(Cache::connection(), 'connection'), 'parameters'), 'parameters'),
    ];
});
```

Above is a route to dump the database.redis config and cache's connection parameters.

Bug version is at branch `master` https://github.com/ifduyue/laravel-5-redis-session-demo/tree/master:

```bash
$ curl http://localhost:8000/ -sS | json_reformat 
{
    "config": {
        "cluster": false,
        "default": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 0
        },
        "for-cache": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 1
        },
        "for-session": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 2
        }
    },
    "cache's connection parameters": {
        "host": "127.0.0.1",
        "port": 6379,
        "database": 2,
        "scheme": "tcp",
        "timeout": 5
    }
}
```

Corrected version is at branch `corrected` https://github.com/ifduyue/laravel-5-redis-session-demo/tree/corrected

```bash
curl http://localhost:8000/ -sS | json_reformat 
{
    "config": {
        "cluster": false,
        "default": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 0
        },
        "for-cache": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 1
        },
        "for-session": {
            "host": "127.0.0.1",
            "port": 6379,
            "database": 2
        }
    },
    "cache's connection parameters": {
        "host": "127.0.0.1",
        "port": 6379,
        "database": 1,
        "scheme": "tcp",
        "timeout": 5
    }
}
```


We can see the difference that in the bug version, cache's connection parameters goes to redis database 2, which is same with the redis cache config. Hence all the Cache operations go into the for-session connection. That's not expected.

**Comment by GrahamCampbell** [comment]
Sorry I wrongly closed this. Just replicated this issue.

**Comment by GrahamCampbell** [comment]
Hmmm, just looked at this code again, and I remember why now. This isn't correct. It's wrong to clone the cache manager.

**Pull request closed without merging**